packages/mobile/src/tokens/saga.ts

```diff
@@ -11,31 +11,71 @@ import { addStandbyTransaction, removeStandbyTransaction } from 'src/transaction
 import { TransactionStatus, TransactionTypes } from 'src/transactions/reducer'
 import { sendAndMonitorTransaction } from 'src/transactions/saga'
 import Logger from 'src/utils/Logger'
-import { web3 } from 'src/web3/contracts'
+import { contractKit, web3 } from 'src/web3/contracts'
 import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
 import * as utf8 from 'utf8'
 
 const TAG = 'tokens/saga'
 
+// The number of wei that represent one unit in a contract
+const contractWeiPerUnit: { [key in CURRENCY_ENUM]: BigNumber | null } = {
+  [CURRENCY_ENUM.GOLD]: null,
+  [CURRENCY_ENUM.DOLLAR]: null,
+}
+
+async function getWeiPerUnit(token: CURRENCY_ENUM) {
+  let weiPerUnit = contractWeiPerUnit[token]
+  if (!weiPerUnit) {
+    const contract = await getTokenContract(token)
+    const decimals = await contract.decimals()
+    weiPerUnit = new BigNumber(10).pow(decimals)
+    contractWeiPerUnit[token] = weiPerUnit
+  }
+  return weiPerUnit
+}
+
+export async function convertFromContractDecimals(value: BigNumber, token: CURRENCY_ENUM) {
+  const weiPerUnit = await getWeiPerUnit(token)
+  return value.dividedBy(weiPerUnit)
+}
+
+export async function convertToContractDecimals(value: BigNumber, token: CURRENCY_ENUM) {
+  const weiPerUnit = await getWeiPerUnit(token)
+  return value.times(weiPerUnit)
+}
+
+export async function getTokenContract(token: CURRENCY_ENUM) {
+  Logger.debug(TAG + '@getTokenContract', `Fetching contract for ${token}`)
+  await getConnectedAccount()
```

Comment:
Why does the account need to be connected for this?

Reply:
Switched to wait for web3 sync, which is needed for contractkit to get the token contracts